src/routes/upload.ts

```diff
@@ -3,14 +3,22 @@ import { os } from "../structs/database";
 import mongoose from "mongoose";
 import { Image } from "../models/image.schema";
 import * as crypto from "crypto";
+import { User } from "../models/user.schema";
 
 const router = Router();
 const bucketName = "";
 
-router.post("/upload", (req, res) => {
-  const userToken = req.body.token;
+router.post("/upload", async (req, res) => {
+  const token = req.body.token;
```

Comment:
Please check if a token was provided: `req.body.hasOwnProperty("token")`

Reply:
And please leave the route empty, the `/upload` endpoint is already provided by the router